src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -277,3 +288,64 @@ def _check_authorization(self, username):
         if not AuthManager.authorize(username, sec_scope, sec_perms):
             raise cherrypy.HTTPError(403, "You don't have permissions to "
                                           "access that resource")
+
+class Oauth2JWTManager(object):
+    _token = ''
+    _token_payload = {}
+    logger = logging.getLogger('oauth2jwt')
+
+
+    @classmethod
+    def set_token(cls, token):
+        cls._token = token
+        cls.set_token_payload()
+        JwtManager.oauth2_token = True
+
+    @classmethod
+    def set_token_payload(cls, token_payload=None):
+        if token_payload:
+            cls._token_payload = token_payload
+        elif cls._token:
+            cls._token_payload = json.loads(base64.urlsafe_b64decode(cls._token.split(".")[1] + "===="))
+
+        if cls._token_payload:
+            JwtManager.oauth2_token = True
+
+    @classmethod
+    def get_token(cls):
+        return cls._token
+
+    @classmethod
+    def _create_user(cls):
+        user_roles = cls.get_user_roles()
+        user = mgr.ACCESS_CTRL_DB.create_user(cls._token_payload['sub'], None, cls._token_payload['name'], cls._token_payload['email'])
+        user.set_roles([Role.map_to_system_roles(user_roles[0])])
+
+        # set user last update to token time issued
+        user.last_update = cls._token_payload['iat']
+        mgr.ACCESS_CTRL_DB.save()
```

Comment:
Why? Why is OAuth2 SSO mingling with the local user DB? The purpose of SSO is to fully offload the user management to a external entity, not to juggle with all the things together. All the user metadata should be kept in some client-side session token (JWT, whatever, ...).

Reply:
The idea with the introduction of this new SSO mechanism was to modify as little as possible the current login, authorization and SSO (SAML2).

The current code uses the JWT token to get the user info and then makes use of the user to check for permission and for it's authorization.
I ended up with this wierd create/delete user, that would be used to authorize in the dashboard.

For now I'm not really sure if I should keep it

Reply:
While drafting the design for the long-term JWT TTLs, I found that the implementation of JWT in the Dashboard is quite peculiar/non-standard (mixing client and server side state). Technically, once the JWT has been granted to the client, the server shouldn't perform any credential (claim) revalidation until JWT expiration.

That said, for OAuth2 we should at least avoid adding/deleting users to the mon DB. That's extra load for the mon DB that was not present with the local auth.